examples/pubsub/tutorial_pubsub_publish_event.c

```diff
@@ -0,0 +1,327 @@
+
+    UA_SimpleAttributeOperand selectedFields [selectedFieldSize];
+    for(size_t i = 0; i < selectedFieldSize; ++i) {
+        UA_SimpleAttributeOperand_init(&selectedFields[i]);
+    }
+
+    selectedFields[0].typeDefinitionId = UA_NODEID_NUMERIC(0, UA_NS0ID_BASEEVENTTYPE);
+    selectedFields[0].attributeId = UA_ATTRIBUTEID_VALUE;
+    selectedFields[0].browsePathSize = 1;
+    UA_QualifiedName browsePathMessage[1];
+    browsePathMessage[0] = UA_QUALIFIEDNAME(0, "Message");
+    selectedFields[0].browsePath = browsePathMessage;
+
+    selectedFields[1].typeDefinitionId = UA_NODEID_NUMERIC(0, UA_NS0ID_BASEEVENTTYPE);
+    selectedFields[1].attributeId = UA_ATTRIBUTEID_VALUE;
+    selectedFields[1].browsePathSize = 1;
+    UA_QualifiedName browsePathSeverity[1];
+    browsePathSeverity[0] = UA_QUALIFIEDNAME(0, "Severity");
+    selectedFields[1].browsePath = browsePathSeverity;
+
+    publishedDataSetConfig.config.event.selectedFields = selectedFields;
+
+    /* Adds a ContentFilter to the PDS*/
+    UA_ContentFilter contentFilter;
+    UA_ContentFilter_init(&contentFilter);
+    UA_ContentFilterElement contentFilterElement;
+    UA_ContentFilterElement_init(&contentFilterElement);
+    UA_ExtensionObject filterOperandExObj;
+    UA_ExtensionObject_init(&filterOperandExObj);
+    UA_LiteralOperand literalOperand;
+    UA_LiteralOperand_init(&literalOperand);
+
+    contentFilter.elementsSize = 1;
+    contentFilter.elements = &contentFilterElement;
+
+    contentFilterElement.filterOperandsSize = 1;
+    contentFilterElement.filterOperands = &filterOperandExObj;
+    contentFilterElement.filterOperator = UA_FILTEROPERATOR_OFTYPE;
+
+    filterOperandExObj.encoding             = UA_EXTENSIONOBJECT_DECODED;
+    filterOperandExObj.content.decoded.type = &UA_TYPES[UA_TYPES_LITERALOPERAND];
+    filterOperandExObj.content.decoded.data = &literalOperand;
+
+    UA_Variant_setScalar(&literalOperand.value, &eventType, &UA_TYPES[UA_TYPES_NODEID]);
+
+    publishedDataSetConfig.config.event.filter = contentFilter;
+
+    /* Create new PublishedDataSet based on the PublishedDataSetConfig. */
+    UA_Server_addPublishedDataSet(server, &publishedDataSetConfig, &publishedDataSetIdent);
+}
+
+
+static void addDataSetFields(UA_Server *server){
+    /*Creates DataSetFields based on the PDS selectedFields*/
+    UA_PublishedDataSet *publishedDataSet = UA_PublishedDataSet_findPDSbyId(server, publishedDataSetIdent);
+    for(size_t i = 0; i < publishedDataSet->config.config.event.selectedFieldsSize; i++){
+        UA_SimpleAttributeOperand sao = publishedDataSet->config.config.event.selectedFields[i];
+        UA_String name = sao.browsePath[0].name;
+
+        UA_NodeId dataSetFieldIdent;
+        UA_DataSetFieldConfig dataSetFieldConfig;
+        memset(&dataSetFieldConfig, 0, sizeof(UA_DataSetFieldConfig));
+        dataSetFieldConfig.dataSetFieldType = UA_PUBSUB_DATASETFIELD_EVENT;
+        dataSetFieldConfig.field.events.fieldNameAlias = name;
```

Comment:
The fieldNameAlias is used in the FieldMetaData and the AddPublishedEvents method has a FieldNameAliases parameter. I agree that we probably don't need the DSF-config at all, we could just add the two fields to the PDS-config. We started out with the DSF-config because we didn't know the exact config that was needed.